app/assets/javascripts/omni-nav-v2.js

```diff
@@ -1,38 +1,109 @@
-var omni_nav_v2 = {
+var OmniNav2 = (function() {
 
-  initialize: function() {
-    this.$container = $('#omni-nav-v2');
-    this.$utility_nav = this.$container.find('.utility-nav');
-    this.$primary_nav = this.$container.find('#primary-nav');
 
+  var $container, $utility_nav, $primary_nav;
 
-    $('.utility-nav-trigger').on('click', omni_nav_v2.onUtilityNavClick);
-    this.$utility_nav.find('li.utility-has-dropdown').on('click', omni_nav_v2.onUtilityNavDropdownClick);
+  var initialize = function(container, utility_nav, primary_nav) {
```

Comment:
Do `utility_nav` and `primary_nav` need to be passed in as arguments? Can they be derived from `container`?